<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>SCRAM - Carroll Lab</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">SCRAM</h1>
      <h2 class="project-tagline">Small Complementary RnA Mapper : a quick and simple small RNA read analysis package</h2>
      <a href="https://github.com/Carroll-Lab/scram" class="btn">View on GitHub</a>
      <a href="https://github.com/Carroll-Lab/scram/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/Carroll-Lab/scram/tarball/master" class="btn">Download .tar.gz</a>
    </section>

    <section class="main-content">
      <h1>
    <a id="scram" class="anchor" href="#scram" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>SCRAM</h1>
        <img src="images/website-image.png" alt="plots" class="inline">
        <p>SCRAM is lightweight Python package for aligning small RNA reads to one
        or more reference sequences and producing publication-quality images. <br><br>
            
            Developed by Stephen Fletcher @ the laboratory of <a href="http://staff.scmb.uq.edu.au/staff/bernard-carroll">Prof. Bernie Carroll</a>, University of Queensland
        </p><br>

    <h3>Installation</h3>
    
        <p>Scram is written in Python 3.5. Install scram and its dependencies with <code>pip</code>:<br><br>
        <code>pip install scram</code><br><br>
        <p>Or download and extract the tarball, then run:<br><br>
        <code>python setup.py install</code><br> <br>
    <h3>Input File Format</h3>
    <p>
    Reference File : DNA nucleotides only (AGCT) - FASTA format
    <br><br>
    Sequence File : Collapsed (unique) reads - DNA nucleotides only (AGCT) - FASTA format
    <br><br>
    Post-processing of FASTQ reads to collapsed FASTA format can be carried out using the 
    <a href="http://hannonlab.cshl.edu/fastx_toolkit/">FASTX-Toolkit from the Hannon Lab</a>.  
     Collapsed reads are unique, and contain the read count in the header.  
    <br><br>
    An example of the required sequence file format:<br>
    <code>
    head seq1.fa<br><br>
    >1-607041<br>
    TCGGACCAGGCATCATTCCCC<br>
    >2-202886<br>
    TCGGACCAGGCTTCATACCCC<br>
    >3-71446<br>
    TCCCAAATATAGACAAAGCA<br>
    </code>

      
    
    <h3>Usage</h3>
        <p>    
        <code>scram analysis_type reference_file [-h] [-s1 [SEQ_FILE_1 [SEQ_FILE_1 ...]]]
             [-s2 [SEQ_FILE_2 [SEQ_FILE_2 ...]]] [-nt SRNA_LEN] [-f FILE_NAME]
             [-p PROCESSES] [-min_read MIN_READ_SIZE]
             [-max_read MAX_READ_SIZE] [-min_count MIN_READ_COUNT]
             [-win SMOOTH_WIN_SIZE] [-ylim YLIM] [-no_csv] [-no_display]
             [-split] [-pub] [-V]</code>
        
        <br><br>
        Analysis types<br>
        <ul>
        <li><code>den</code> : align reads of a single sRNA class (eg. 21 nt) from one or more replicate sequence files to a single reference sequence
        (<code>-s1</code> and <code>-nt</code> required)</li>
        <li><code>mnt3dm</code> : align 21, 22 and 24 nt reads from one or more replicate sequence files to a single reference sequence
        (<code>-s1</code> required)</li>
        <li><code>CDP</code> : count aligned reads of a single sRNA class (eg. 21 nt) to multiple reference sequences. Counts for two sets of one or more replicate sequence files are plotted as (x,y) coordinates for each reference
        (<code>-s1</code>, <code>-s2</code> and <code>-nt</code> required)</li>
        <li><code>CDP_single</code> : count aligned reads of a single sRNA class (eg. 21 nt) to multiple reference sequences. Counts are written to a .csv file with a column for each read file.  No plot output.
        (<code>-s1</code> and <code>-nt</code> required)</li>   
          
        </ul>
        Flags<br>
        <ul>
          <li><code>-h</code> : Help message
          <li><code>-s1</code> : Sequence file 1 (if more than 1 replicate sequence file, read counts are averaged)</li>
                                
          <li><code>-s2</code> : Sequence file 2 (if more than 1 replicate sequence file, read counts are averaged)</li>
                                
          <li><code>-nt</code> : sRNA length to analyse</li>
          
          <li><code>-p</code> : number of processes (CPU cores) to use in CDP analyses (default=4)</li>          
          
          <li><code>-f</code> : Figure output file name (if not auto-generated). Use 'auto' to auto-generate</li>
                                
          <li><code>-min_read</code> : Minimum length of sRNA reads used for normalisation (default=18) </li>
                                
          <li><code>-max_read</code> : Maximum length of sRNA reads used for normalisation (default=32)</li>
                                
          <li><code>-min_count</code> : Minimum read count for an sRNA to be aligned and used for normalisation
                                (default=1)</li>

          <li><code>-win</code> :  Window size for smoothing of den plots (default=50, min = 6)</li>

          <li><code>-ylim</code> : +/- y-axis limit on plots</li>
                                
          <li><code>-no_display</code> : Do not display plot on screen</li>
          <li><code>-no_csv</code> : Do not generate the .csv alignment file</li>        
                                
          <li><code>-split</code> : Split read alignment counts based on no. of
                                alignments (i.e. if a read aligned in 2 positions, the read count at each position is halved)</li>
                                
          <li><code>-pub</code> : Remove all axis labels and legends for preperation of figures for publication</li>
          <li><code>-V</code> : Show program's version number and exit</li>
        </ul>
    <h3>den Example</h3>
        <code> scram den ./ref.fa -s1 seq1.fa -nt 24 -win 30 -f fig1.pdf</code><br>
        <img src="images/example-1-den.png" alt="example 1" class="inline" style="width:600px;height:400px;"><br>
        
    <h3>mnt3dm Example</h3>   
        <code> scram mnt3dm ./ref.fa -s1 seq1.fa -win 20 -ylim 110 -f fig2.pdf</code><br><br><br>
        <img src="images/example2-mnt3dm.png" alt="example 2" class="inline" style="width:600px;height:400px;"><br>
        
    <h3>CDP Example</h3>    
        <code> scram CDP ./cDNAs.fa -s1 seq1.fa -s2 seq2.fa -nt 21 -f fig3.pdf -split</code><br>
        <img src="images/example3-CDP.png" alt="example 3" class="inline" style="width:600px;height:400px;"><br><br>
      
    <br><br><br>(c) 2016 - Stephen Fletcher.   <a href="https://opensource.org/licenses/MIT">MIT License</a>   
      
      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/Carroll-Lab/scram">Scram</a> is maintained by <a href="https://github.com/Carroll-Lab">Carroll-Lab</a>.</span>

        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
      </footer>

    </section>

  
  </body>
</html>
