<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>SCRAM - Carroll Lab</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">SCRAM</h1>
      <h2 class="project-tagline">Small Complementary RnA Mapper : a quick and simple small RNA read analysis package</h2>
      <a href="https://github.com/Carroll-Lab/scram" class="btn">View on GitHub</a>
      <a href="https://github.com/Carroll-Lab/scram/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/Carroll-Lab/scram/tarball/master" class="btn">Download .tar.gz</a>
    </section>

    <section class="main-content">
      <h1>
    <a id="scram" class="anchor" href="#scram" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>SCRAM</h1>
        <img src="images/website-image.png" alt="plots" class="inline">
        <p>SCRAM is lightweight Python package for aligning small RNA reads to one
        or more reference sequences and producing publiciation-quality images. <br><br>
            
            Developed by Stephen Fletcher @ the laboratory of <a href="http://staff.scmb.uq.edu.au/staff/bernard-carroll">Prof. Bernie Carroll</a>, University of Queensland
        </p>

    <h3>Installation</h3>
        <p>Unpack to a location of your choice. Use directly :<br>
        <br><code>python scram.py req1 req 2 [opt]</code><br>
            
        <br>Or install in your path : <br>
        <ol>
        <li>If your Python 2.7 path differs from <code>/usr/bin/python</code>  - modify line 1 of scram.py to reflect
         your python path.
         <li>Make scram.py executable : <code>chmod +x /path/to/scram.py</code>
         </ol>
             
         <code>scram.py req1 req 2 [opt]</code><br><br>
        Required Python packages:<br><br>
        <code>matplotlib</code>
        <br>
        <code>numpy</code>
        </p>  
    <h3>File Format</h3>
    <p>
    Reference File - DNA sequences only (AGCT) - FASTA format
    <br>
    Sequence File - Collapsed reads - DNA sequences only (AGCT) - FASTA format
    <br>
    Post-processing of FASTQ reads to collapsed FASTA foramt can be carried out using the 
    <a href="http://hannonlab.cshl.edu/fastx_toolkit/">FASTX-Toolkit from the Hannon Lab</a>.
    <br>
    An example of the required format:
    <code>
    head seq1.fa<br>
    >1-607041
    TCGGACCAGGCATCATTCCCC
    >2-202886
    TCGGACCAGGCTTCATACCCC
    >3-71446
    TCCCAAATATAGACAAAGCA
    </code>

      
    
    <h3>Usage</h3>
        <p>    
        <code>scram.py analysis_type reference_file [-h] [-s1 SEQ_FILE_1] [-s2 SEQ_FILE_2] [-s3 SEQ_FILE_3]
                [-s4 SEQ_FILE_4] [-nt SRNA_LEN] [-f FILE_NAME]
                [-seq_list SEQ_LIST] [-min_read MIN_READ_SIZE]
                [-max_read MAX_READ_SIZE] [-min_count MIN_READ_COUNT]
                [-win SMOOTH_WIN_SIZE] [-ylim YLIM] [-no_display] [-split]
                [-pub] [-V]</code>
        </p>
      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/Carroll-Lab/scram">Scram</a> is maintained by <a href="https://github.com/Carroll-Lab">Carroll-Lab</a>.</span>

        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
      </footer>

    </section>

  
  </body>
</html>
