<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>SCRAM - Carroll Lab</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">SCRAM</h1>
      <h2 class="project-tagline">Small Complementary RnA Mapper : a quick and simple small RNA read analysis package</h2>
      <a href="https://github.com/Carroll-Lab/scram" class="btn">View on GitHub</a>
      <a href="https://github.com/Carroll-Lab/scram/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/Carroll-Lab/scram/tarball/master" class="btn">Download .tar.gz</a>
    </section>

    <section class="main-content">
      <h1>
    <a id="scram" class="anchor" href="#scram" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>SCRAM</h1>
        <img src="images/website-image.png" alt="plots" class="inline">
        <p>SCRAM is lightweight Python package for aligning small RNA reads to one
        or more reference sequences and producing publication-quality images. <br><br>
            
            Developed by Stephen Fletcher @ the laboratory of <a href="http://staff.scmb.uq.edu.au/staff/bernard-carroll">Prof. Bernie Carroll</a>, University of Queensland
        </p><br>

    <h3>Installation</h3>
        <p>Unpack to a location of your choice. Use directly :<br>
        <br><code>python path/to/scram analysis reference [opt]</code><br>
            
        <br>Or install in your path : <br>
        <ol>
        <li>If your Python 2.7 path differs from <code>/usr/bin/python</code>  - modify line 1 of scram to reflect
         your python path.
         <li>Make scram executable : <code>chmod +x /path/to/scram</code>
         </ol>
             
         <code>scram analysis reference [opt]</code><br><br>
        Required Python packages:<br><br>
        <code>matplotlib</code>
        <br>
        <code>numpy</code><br> 
    <h3>Input File Format</h3>
    <p>
    Reference File : DNA nucleotides only (AGCT) - FASTA format
    <br><br>
    Sequence File : Collapsed reads - DNA nucleotides only (AGCT) - FASTA format
    <br><br>
    Post-processing of FASTQ reads to collapsed FASTA format can be carried out using the 
    <a href="http://hannonlab.cshl.edu/fastx_toolkit/">FASTX-Toolkit from the Hannon Lab</a>.  
     Collapsed reads are unique, and contain the read count in the header.  
    <br><br>
    An example of the required sequence file format:<br>
    <code>
    head seq1.fa<br><br>
    >1-607041<br>
    TCGGACCAGGCATCATTCCCC<br>
    >2-202886<br>
    TCGGACCAGGCTTCATACCCC<br>
    >3-71446<br>
    TCCCAAATATAGACAAAGCA<br>
    </code>

      
    
    <h3>Usage</h3>
        <p>    
        <code>scram analysis_type reference_file [-h] [-s1 SEQ_FILE_1] [-s2 SEQ_FILE_2] [-s3 SEQ_FILE_3]
                [-s4 SEQ_FILE_4] [-nt SRNA_LEN] [-f FILE_NAME]
                [-seq_list SEQ_LIST] [-min_read MIN_READ_SIZE]
                [-max_read MAX_READ_SIZE] [-min_count MIN_READ_COUNT]
                [-win SMOOTH_WIN_SIZE] [-ylim YLIM] [-no_display] [-split]
                [-pub] [-V]</code>
        
        <br><br>
        Analysis types<br>
        <ul>
        <li><code>den</code> : align reads of a single sRNA class (eg. 21 nt) from a single sequence file to a single reference sequence
        (<code>-s1</code> and <code>-nt</code> required)</li>
        <li><code>denAv</code> : align reads of a single sRNA class (eg. 21 nt) to a single reference. Reads are the average from 2 replicate sequence files
        (<code>-s1</code>, <code>-s2</code> and <code>-nt</code> required)</li>
        <li><code>mnt3dm</code> : align 21, 22 and 24 nt reads from a single sequence file to a single reference sequence
        (<code>-s1</code> required)</li>
        <li><code>multiDen</code> : align reads of a single sRNA class (eg. 21 nt) from a multiple sequence files to multiple reference sequences
        (<code>-seq_list</code> required)</li>
        <li><code>multiDen</code> : align reads of a single sRNA class (eg. 21 nt) from a multiple replicate sequence files to multiple reference sequences
        (<code>-seq_list</code> required)</li>
        <li><code>CDP</code> : count aligned reads of a single sRNA class (eg. 21 nt) to multiple reference sequences. Counts for two sequence files are plotted as (x,y) coordinates for each reference
        (<code>-s1</code>, <code>-s2</code> and <code>-nt</code> required)</li>
        <li><code>avCDP</code> : count aligned reads of a single sRNA class (eg. 21 nt) to multiple reference sequences. Counts for two replicate sequence files are plotted as (x,y) coordinates for each reference
        (<code>-s1</code>, <code>-s2</code>, <code>-s3</code>, <code>-s4</code>and <code>-nt</code> required)</li>
        <li><code>sCDP</code> : count aligned reads of a single sRNA class (eg. 21 nt) to multiple reference sequences. Single sequence file required. CSV output only.
        (<code>-s1</code> and <code>-nt</code> required)</li>             
        </ul>
        Flags<br>
        <ul>
          <li><code>-h</code> : Help message
          <li><code>-s1</code> : Sequence file 1</li>
                                
          <li><code>-s2</code> : Sequence file 2</li>
        
          <li><code>-s3</code> : Sequence file 3</li>
                                
          <li><code>-s4</code> : Sequence file 4</li>
                                
          <li><code>-nt</code> : sRNA length to analyse</li>
          <li><code>-f</code> : Figure output file name (if not auto-generated)</li>
                                
          <li><code>-seq_list</code> : Text (.txt) file with full path of sequence file on each line (single replicate) or two tab-delimited sequence file paths per line (two replicates)</li>
                                
          <li><code>-min_read</code> : Minimum length of sRNA reads used for normalisation (default=18) </li>
                                
          <li><code>-max_read</code> : Maximum length of sRNA reads used for normalisation (default=32)</li>
                                
          <li><code>-min_count</code> : Minimum read count for an sRNA to be aligned and used for normalisation
                                (default=1)</li>

          <li><code>-win</code> :  Window size for smoothing of den plots (default=50)</li>

          <li><code>-ylim</code> : +/- y-axis limit on plots</li>
                                
          <li><code>-no_display</code> : Do not display plot on screen</li>
                                
          <li><code>-split</code> : Split CDP read alignment counts based on no. of
                                alignments</li>
                                
          <li><code>-pub</code> : Remove all labels from density maps for publication</li>
          <li><code>-V</code> : Show program's version number and exit</li>
        </ul>
    <h3>Examples</h3>
        <code> scram den ./ref.fa -s1 seq1.fa -nt 24 -win 30 -f fig1.pdf</code><br><br>
        <img src="images/example-1-den.png" alt="example 1" class="inline" style="width:304px;height:228px;"><br><br>
        <code> scram mnt3dm ./ref.fa -s1 seq1.fa -win 20 -ylim 110 -f fig2.pdf</code><br><br>
        <img src="images/example2-mnt3dm.png" alt="example 2" class="inline" style="width:304px;height:228px;"><br><br>
        <code> scram CDP ./cDNAs.fa -s1 seq1.fa -s2 seq2.fa -nt 21 -f fig3.pdf -split</code><br><br>
        <img src="images/example3-CDP.png" alt="example 3" class="inline" style="width:304px;height:228px;"><br><br>
      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/Carroll-Lab/scram">Scram</a> is maintained by <a href="https://github.com/Carroll-Lab">Carroll-Lab</a>.</span>

        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
      </footer>

    </section>

  
  </body>
</html>
